llvm/lib/Transforms/Scalar/EarlyCSE.cpp

```diff
@@ -1588,6 +1589,18 @@ bool EarlyCSE::processNode(DomTreeNode *Node) {
         if (InVal.IsLoad)
           if (auto *I = dyn_cast<Instruction>(Op))
             combineMetadataForCSE(I, &Inst, false);
+
+        // If the load has align metadata, preserve it via an alignment
+        // assumption. Note that this doesn't use salavageKnowledge, as we need
+        // to create the assumption for the value we replaced the load with.
+        if (auto *AlignMD = Inst.getMetadata(LLVMContext::MD_align)) {
```

Comment:
You also need `!noundef` for this transform. `!align` by itself only returns poison, while the assumption converts it into IUB.

Reply:
Should be updated, thanks!